eras/shelley/impl/testlib/Test/Cardano/Ledger/Shelley/ImpTest.hs

```diff
@@ -910,6 +949,30 @@ instance MonadWriter [SomeSTSEvent era] (ImpTestM era) where
     ((a, f), evs) <- listen act
     writer (a, f evs)
 
+impEventsFrom ::
+  ImpTestM era () ->
+  ImpTestM era [SomeSTSEvent era]
+impEventsFrom = fmap (toList . snd) . listen
+
+impTransactionsFrom ::
+  ImpTestM era () ->
+  ImpTestM era (StrictSeq (Tx TopTx era))
+impTransactionsFrom act = do
+  mTxsPrev <- use impRecordedTransactionsL
+  impRecordedTransactionsL .= SJust mempty
+  act
+  mTxsDuring <- use impRecordedTransactionsL
+  impRecordedTransactionsL .= liftA2 (<>) mTxsPrev mTxsDuring
+  pure $ fold mTxsDuring
+
+impWithoutRecordingTxs :: ImpTestM era a -> ImpTestM era a
+impWithoutRecordingTxs act = do
+  prev <- use impRecordedTransactionsL
+  impRecordedTransactionsL .= SNothing
+  res <- act
+  impRecordedTransactionsL .= prev
+  pure res
```

Comment:
There should be never a use for this function. Why would one submit a transaction, modify the state and then discard the transaction and continue recording other transactions? This would at the very least totally mess up the fixup functionality.

Let' get rid of it. If it will be ever useful, then we can add it, at least with some strange use case in mind.